tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -17,35 +18,1258 @@
 #include <zephyr/bluetooth/audio/pacs.h>
 
 #include "bap_endpoint.h"
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_bap
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "btp/btp.h"
 
+#define DISCOVERY_TIMEOUT 300
+
+#define DEFAULT_CONTEXT BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | BT_AUDIO_CONTEXT_TYPE_MEDIA
 #define SUPPORTED_SINK_CONTEXT	(BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
 				 BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
 				 BT_AUDIO_CONTEXT_TYPE_MEDIA | \
 				 BT_AUDIO_CONTEXT_TYPE_GAME | \
 				 BT_AUDIO_CONTEXT_TYPE_INSTRUCTIONAL)
 
-#define SUPPORTED_SOURCE_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
-				  BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
-				  BT_AUDIO_CONTEXT_TYPE_MEDIA | \
-				  BT_AUDIO_CONTEXT_TYPE_GAME)
+#define SUPPORTED_SOURCE_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+				  BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
+				  BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+				  BT_AUDIO_CONTEXT_TYPE_GAME)
+
+#define AVAILABLE_SINK_CONTEXT SUPPORTED_SINK_CONTEXT
+#define AVAILABLE_SOURCE_CONTEXT SUPPORTED_SOURCE_CONTEXT
+
+static struct bt_codec default_codec =
+	BT_CODEC_LC3(BT_CODEC_LC3_FREQ_ANY, BT_CODEC_LC3_DURATION_10,
+		     BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1),
+		     40u, 120u, 1u,
+		     (BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL |
+		      BT_AUDIO_CONTEXT_TYPE_MEDIA));
+
+struct audio_stream {
+	struct bt_bap_stream stream;
+	uint16_t seq_num;
+	uint16_t max_sdu;
+	size_t len_to_send;
+};
+
+#define MAX_STREAMS_COUNT MAX(CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT, \
+			     CONFIG_BT_ASCS_ASE_SNK_COUNT) + MAX(CONFIG_BT_ASCS_ASE_SRC_COUNT,\
+			     CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT)
+#define MAX_END_POINTS_COUNT CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT + \
+			     CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT
+struct audio_connection {
+	struct bt_conn *conn;
+	struct audio_stream streams[MAX_STREAMS_COUNT];
+	size_t configured_sink_stream_count;
+	size_t configured_source_stream_count;
+	struct bt_codec codec;
+	struct bt_codec_qos qos;
+	struct bt_bap_unicast_group *unicast_group;
+	struct bt_bap_ep *end_points[MAX_END_POINTS_COUNT];
+	size_t end_points_count;
+	uint8_t chan_id;
+} connections[CONFIG_BT_MAX_CONN];
+uint8_t next_chan_id = 1;
+
+static struct bt_bap_unicast_client_discover_params ase_discover_params;
+
+static struct bt_codec_qos_pref qos_pref = BT_CODEC_QOS_PREF(true, BT_GAP_LE_PHY_2M, 0x02,
+							     10, 40000, 40000, 40000, 40000);
```

Comment:
```suggestion
							     10, 10000, 40000, 10000, 40000);
```

Suggest to have a proper range between min and max PD